mkFit/HitStructures.h

```diff
@@ -48,6 +48,10 @@ typedef std::array<PhiBinInfo_t, Config::m_nphi> vecPhiBinInfo_t;
 
 typedef std::vector<vecPhiBinInfo_t> vecvecPhiBinInfo_t;
 
+typedef std::array<bool, Config::m_nphi> vecPhiBinDead_t;
```

Comment:
Config::m_nphi = 128, 6.28/128 = .04906

Should we consider trying to increase phi-bins to 256 to reduce false identification of bad hits? IIRC, Slava said there is little speed change if one varies this variable. It was 1024 before Slava's last changes here.

Reply:
at a radius of 1.1 m (roughly the last layer, TOB) 128 bins corresponds to 5.4 cm.
TOB modules are 9 cm. So, a phi bin is close to a half of the module or less.
TIB modules are 6 cm (the last TIB radius is 50 cm; a phi bin size at this radius is 2.5 cm).
BPIX is 1.6 cm (the last layer is 16 cm; a bin size is 0.8 cm).

So, unless this technology is going to appropriately pick up fractions of modules we will just be repeating the values.

Reply:
Since the overall structure of the bins is already in place, repeating values means just a bool and is thus a very small effect. So it's good that the bin size is smaller than the module size, I was more concerned that we could over-assign invalid hits. To me this means that the dead modules do not require an increase in phi bins, but if we choose to do so that is not going to create problems either

Reply:
> repeating values means just a bool and is thus a very small effect

Just to note that `std::array<bool, N>` takes one byte per element (in case with you were after "bit per element" with "just a bool"). Nevertheless a byte per element could be ok storage-wise (128 bytes per layer if I guess correctly?), and avoids bit shifts (although their real cost in modern hardware is unclear to me).

Reply:
This is actually even more pronounced for q-bins. We have both a) too large q bins (at least in some layers), and b) earch an extra q-bin up/down. So we effectively pick up dead regions way too often.

Maybe we could restrict this matching at least in q, unless the track passes relatively close to the edge of the bin.

We should re-check q-bins as well. @mmasciov haven't I tried to stick it on you a couple of months back? :)

Reply:
And it's 128 bytes * N_q_bins. We should switch to bits ... that part of code is heavily L1 limited.